### PR TITLE
Add booking email with ics attachment

### LINKED ISSUE #3855: Send event invitation to the person who booked an appointment
**Is your feature request related to a problem? Please describe.**
After clicking the confirmation link for an appointment, I'd expect to be mailed a proper event invitation to add to my local calendar.

Currently this doesn't happen.

**Describe the solution you'd like**
See above.

**Describe alternatives you've considered**
Addding the event to the confirmation email as described here: https://github.com/nextcloud/calendar/issues/3852#issuecomment-1002146079

This would probably have way more potential for confusion if users forget to also click the confirmation link.

**Additional context**
See #3852

### PR DESCRIPTION
**Pull request by miaulalala**
Fixes #3855

| Before  | After |
| ------------- | ------------- |
| Subject: Your appointment "Have Lunch With Me" needs confirmation | Subject: Your appointment "Have Lunch With Me" with admin needs confirmation |
| image  |  image |

New booking info email:
image

### CODE REVIEW

**Comment by miaulalala** [comment]
Maybe extend the text a bit... IDK, the text isn't very descriptive at the moment.

File: lib/Service/Appointments/BookingCalendarWriter.php [hunk 1]
```diff
@@ -38,6 +38,7 @@
 use OCP\Security\ISecureRandom;
 use RuntimeException;
 use Sabre\VObject\Component\VCalendar;
+use Sabre\VObject\InvalidDataException;
```
Comment (ChristophWurst) [inline lib/Service/Appointments/BookingCalendarWriter.php:41]: :sparkles:

**Review by ChristophWurst — APPROVED** [review]

**Review by st3iny — APPROVED** [review]

File: lib/Service/Appointments/MailService.php [hunk 2]
```diff
@@ -144,11 +149,77 @@ public function sendConfirmationEmail(Booking $booking, AppointmentConfig $confi
...
+
+		$template->addBodyListItem($user->getDisplayName(), 'Appointment with:');
+		if (!empty($config->getDescription())) {
+			$template->addBodyListItem($config->getDescription(), 'Description:');
+		}
+
+		// Create Booking overview
+		$this->addBulletList($template, $this->l10n, $booking, $config->getLocation());
+
+		$bodyText = $this->l10n->t('If you wish to cancel the appointment after all, please contact your organizer.');
```
Comment (tcitworld) [inline lib/Service/Appointments/MailService.php:193]: Maybe add something like « by replying to this email or heading to their profile page » (first part should check if `$fromEmail` isn't `null`).

**Review by tcitworld — APPROVED** [review]

File: lib/Service/Appointments/MailService.php (hunk 2 above)
Comment (miaulalala) [reply lib/Service/Appointments/MailService.php:193]: from email is sys email. but I could add a reply to header.

Comment (tcitworld) [reply lib/Service/Appointments/MailService.php]: I meant when the organizer has no email set, but I had forgotten that can't happen. :)

**Comment by JMoVS** [comment]
@miaulalala nice work! But is there an option to skip the email verification step?

**Comment by miaulalala** [comment]
> @miaulalala nice work! But is there an option to skip the email verification step?

Hm it's a feature to think about maybe. But for now, no, there isn't. You can open a feature request on this repo, then we have that documented as a ticket.

**Pull request merged**